### PR TITLE
Fix part of #4044: Add accessibility string generation support for math expressions

### PR DESCRIPTION
**Pull request by BenHenning**
## Explanation
Fix part of #4044
Originally copied from #2173 when it was in proof-of-concept form

This PR introduces a utility for generation human-readable accessibility strings from math expressions and equations, per [this document](https://docs.google.com/document/d/1SkzAD4k7SWLp5_3L5WNxsnR79ATlOk8pz4irfE2ls-4/edit?usp=sharing). This is needed since the LaTeX and raw representations of expressions aren't guaranteed to be read correctly by screenreaders.

Broadly, this composes an sentence from fragments corresponding to each part of the expression. Currently, only English is supported since a custom routine is needed for each language. See the tests for ideas on the supported scenarios.

MathExpressionAccessibilityUtilTest is being exempted for parameterization since there's an advantage to covering a lot of different conditions for the same base behaviors. This is slightly abusing parameterization since the tests including expected outcomes, but it seems sensible given how the tests are being organized.

Further, TestActivity was updated so that MathExpressionAccessibilityUtil can be tested with a display locale. The resource handler & display locale were updated to include long/double format methods since we can't rely on regular toString() methods since they'll introduce scientific notation (for doubles) or potentially be treated as phone numbers (for longs). This formatting is localized base on the user's locale despite the generated strings always being in English.

Note that overall this is somewhat an accessibility stop-gap since some of the strings are quite long and likely difficult to fully contextualize for a screenreader user. Ideally, we'd introduce some sort of navigation flow longer term so that the learner can read out specific parts of the expression in a more pedagogically friendly way. This should definitely be possible to build on top of the existing infrastructure being introduced in this & past PRs.

## Essential Checklist
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
This isn't hooked up to any UIs yet, and a later PR will demonstrate the end-to-end flow (including accessibility). See MathExpressionAccessibilityUtilTest for example test cases to get an idea on possible accessibility strings.

### CODE REVIEW

**Comment by BenHenning** [comment]
Hi. As of today, some main reviewers have taken time off for the next few weeks, so it may take a little while before we can look at this PR. We appreciate your patience while some of our team members recharge. We'll be fully returning on 4 January 2021.

File: app/src/main/java/org/oppia/android/app/utility/math/MathExpressionAccessibilityUtil.kt [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    return if (lhsStr != null && rhsStr != null) {
+      resourceHandler.getStringInLocaleWithWrapping(
+        R.string.math_accessibility_a_equals_b, lhsStr, rhsStr
+      )
+    } else null
+  }
+
+  private fun MathExpression.toHumanReadableEnglishString(divAsFraction: Boolean): String? {
+    // Reference:
+    // https://docs.google.com/document/d/1P-dldXQ08O-02ZRG978paiWOSz0dsvcKpDgiV_rKH_Y/view.
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/utility/math/MathExpressionAccessibilityUtil.kt]: This isn't the correct link. Reminder to fix this before submitting (& any other references in code or the PR description).

**Comment by BenHenning** [comment]
@rt4914 PTAL for codeowners.
@anandwana001 PTAL at the whole PR as part of the broader math expressions project.

**Comment by rt4914** [comment]
Will review this tonight or tomorrow morning.

**Review by rt4914 — APPROVED** [review]
Looks very clean and nice solution. LGTM.

**Comment by BenHenning** [comment]
This is still active.

**Review by anandwana001 — APPROVED** [review]
LGTM

File: app/src/main/java/org/oppia/android/app/utility/math/MathExpressionAccessibilityUtil.kt (hunk 1 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/utility/math/MathExpressionAccessibilityUtil.kt]: Both this & the PR description have been fixed.

**Comment by BenHenning** [comment]
Note to reviewers: the code & PR description now point to the correct reference document if you'd like to take a look at that (in case any context was missing during the review).

**Pull request merged**